scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,375 @@
+                        .replace(" Plus ", "Or")\
+                        .replace("Armed Forces Overseas", \
+                            "InUSArmedForcesOverseas")\
+                        .replace("Household", \
+                                "ResidesInHousehold")\
+                        .replace("Resident", "USResident")\
+                        .replace("Noninstitutionalized", \
+                            "NonInstitutionalized")\
+                        .strip()\
+                        .replace(" ", "_")\
+                        .replace("__", "_")\
+                        for col in final_cols]
+
+        final_cols_list = computed_cols + final_cols_list
+        df.columns = final_cols_list
+
+        # Multiplying the data with scaling factor.
+        for col in final_cols_list:
+            if "count" in col.lower():
+                if self.scaling_factor != 1:
+                    df[col] = df[col].astype('float', errors="ignore").multiply(
+                        self.scaling_factor)
+                df[col] = df[col].astype('Int64', errors="ignore")
+        self.scaling_factor = 1
+
+        # Creating Location column with default value country/USA.
+        # as the dataset is all about USA country level only.
+        df.insert(1, "Location", "country/USA", True)
+        df.insert(0, 'date_range', _year_range(df['Date']), True)
+        return df
+
+    def _transform_data(self, df: pd.DataFrame) -> None:
+        """
+        This method calls the required functions to transform
+        the dataframe and saves the final cleaned data in
+        CSV file format.
+
+        Arguments:
+            file (str) : Dataset File Path
+
+        Returns:
+            df (DataFrame) : DataFrame.
+        """
+
+        # Finding the Dir Path
+        file_dir = os.path.dirname(self.cleaned_csv_file_path)
+        if not os.path.exists(file_dir):
+            os.mkdir(file_dir)
+        df = self._transform_df(df)
+        keep_value = "first"
+        if self.df is None:
+            self.df = df
+        else:
+            self.df = self.df.append(df, ignore_index=True)
+
+        self.df.sort_values(by=['Date', 'date_range'],
+                            ascending=False,
+                            inplace=True)
+        self.df.drop_duplicates("Date", keep=keep_value, inplace=True)
+        self.df.drop(['date_range'], axis=1, inplace=True)
+        self.df.to_csv(self.cleaned_csv_file_path, index=False)
+
+    def process(self):
+        """
+        This is main method to iterate on each file,
+        calls defined methods to clean, generate final
+        cleaned CSV file, MCF file and TMCF file.
+        """
+        for file in self.input_files:
+            df = self._load_data(file)
+            self._transform_data(df)
+        self._generate_mcf(self.df.columns)
+        self._generate_tmcf(self.df.columns)
+
+    def _generate_mcf(self, df_cols: list) -> None:

```

Comment:
Reorder process, generate_mcf and generate_tmcf

Reply:
reordered for process to be at the end